In [9]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

print("Libraries imported successfully!")

Libraries imported successfully!


In [10]:
data = {
    "StudyHours": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "ExamScore": [35, 40, 45, 50, 55, 60, 65, 72, 80, 88]
}

df = pd.DataFrame(data)

print("Dataset:")
display(df)

Dataset:


,StudyHours,ExamScore
0,1,35
1,2,40
2,3,45
3,4,50
4,5,55
5,6,60
6,7,65
7,8,72
8,9,80
9,10,88


In [11]:
X = df[["StudyHours"]]
y = df["ExamScore"]

print("X shape:", X.shape)
print("y shape:", y.shape)

X shape: (10, 1)
y shape: (10,)


In [12]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

print("Training data:", len(X_train))
print("Testing data:", len(X_test))

Training data: 8
Testing data: 2


In [13]:
model = LinearRegression()

model.fit(X_train, y_train)

print("Linear Regression model trained successfully!")

Linear Regression model trained successfully!


In [14]:
y_pred = model.predict(X_test)

print("Actual Scores:")
print(y_test.values)

print("Predicted Scores:")
print(y_pred)

Actual Scores:
[80 40]
Predicted Scores:
[78.72413793 38.77586207]


In [15]:
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("Mean Squared Error (MSE):", mse)
print("R² Score:", r2)

Mean Squared Error (MSE): 1.563168846611168
R² Score: 0.9960920778834721


In [16]:
print("Coefficient:", model.coef_[0])
print("Intercept:", model.intercept_)

Coefficient: 5.706896551724139
Intercept: 27.36206896551724


In [17]:
print("Prediction Equation:")
print(f"ExamScore = {model.coef_[0]:.2f} × StudyHours + {model.intercept_:.2f}")

Prediction Equation:
ExamScore = 5.71 × StudyHours + 27.36


## Conclusion

A Simple Linear Regression model was successfully developed to predict Exam Score based on Study Hours.

The model was evaluated using Mean Squared Error (MSE) and R² Score.

The prediction equation obtained was:

ExamScore = 5.71 × StudyHours + 27.36

The coefficient indicates that an increase of one study hour is associated with an increase of approximately 5.71 marks in the predicted exam score.

Overall, the task successfully demonstrated the basic implementation and evaluation of Simple Linear Regression using Python and Scikit-learn.